### DISCUSSION TITLE
Python3のscript実行時のdisk I/O errorについて

### DISCUSSION THREAD

**Discussion by hhibara** [nig-sc/nigsc_homepage2#423] [Q&A] [answered]
OS update前に実行できたscriptが実行できません。
自分のhomeの中のファイルは大量に削除しました。
数回scriptを実行したところ、jobを投入してからエラーで終了するまでの時間はまちまちです。
sgelogを見ても、どの行でエラーになるかは、まちまちです。
エラーを回避する方法をご存知でしたら、ご教授いただけますと幸いです。よろしくお願いします。

* 最新のqsub投入時のsgelogのe 7行
Traceback (most recent call last):
  File "/home/hibara/Work/2024/01-Tae-r/scripts/x-01-create-database.py", line 277, in <module>
    _main()
  File "/home/hibara/Work/2024/01-Tae-r/scripts/x-01-create-database.py", line 29, in _main
    db.execute('CREATE TABLE study(study_accession TEXT PRIMARY KEY, study_type TEXT, study_title TEXT, study_abstract TEXT, study_description TEXT, study_links TEXT, bio_project TEXT, geo_id TEXT)')
sqlite3.OperationalError: disk I/O error

* 昨日qsub投入時のsgelogのe 8728行
先頭から
disk I/O error
Traceback (most recent call last):
  File "/home/hibara/Work/2024/01-Tae-r/scripts/x-01-create-database.py", line 44, in _main
    process_experiment(db, os.path.join(root, one), os.path.basename(root))
  File "/home/hibara/Work/2024/01-Tae-r/scripts/x-01-create-database.py", line 135, in process_experiment
    db.execute('INSERT INTO experiment(experiment_accession, study_accession, sample_accession, experiment_title, design_description, platform, platform_model, library_name, library_strategy, library_source, library_selection, library_construction_protocol, spot_length, read_specs, submission_accession) VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)',
sqlite3.OperationalError: disk I/O error
以下続く

**Comment by sc-fs** [comment, marked as answer]
エラーは空き容量が無いのが原因と思われます。
アカウントのホーム領域を確認しましたが、以下のように空き容量が無い状態です。
リソース拡張するか、再度、不要ファイルを削除するようお願いします。
```
$ lfs quota -u hibara /lustre7
Disk quotas for usr hibara (uid 5527):
     Filesystem  kbytes   quota   limit   grace   files   quota   limit   grace
       /lustre7 19999975568       0 20000000000       - 1984420       0       0       -
uid 5527 is using default file quota setting
$#
```
まだ、以下に16Tほどのファイルがあるようです。
`16T     /home/hibara/Work/2023/11-Bowtie2_k5/tmp_k5`

**Reply by hhibara** [reply]
ご確認ありがとうございます。
早速、不要ファイルの削除を行います。
